app/gui/src/main/scala/org/bitcoins/gui/WalletGUIModel.scala

```diff
@@ -23,23 +25,22 @@ class WalletGUIModel() {
   lazy val parentWindow: ObjectProperty[Window] =
     ObjectProperty[Window](null.asInstanceOf[Window])
 
-  private case object UpdateBalanceRunnable extends Runnable {
+  val system: ActorSystem = ActorSystem(
```

Comment:
Can we take in the system as a parameter? So we can share the system across gui + backend

i.e. 

```scala
class WalletGUIModel()(implicit system: ActorSystem) { ... } 
```

That way we don't have two redundant thread pools, and in the future we might use akka's event bus to propagate updates to the GUI